rafthttp/transport.go

```diff
@@ -78,13 +81,25 @@ type Transporter interface {
 	Stop()
 }
 
-type transport struct {
-	roundTripper http.RoundTripper
-	id           types.ID
-	clusterID    types.ID
-	raft         Raft
-	serverStats  *stats.ServerStats
-	leaderStats  *stats.LeaderStats
+// Transport implements Transporter interface. It provides the functionality
+// to send raft messages to peers, and receive raft messages from peers.
+// User should let the handler returned by Handler method in Transport
+// serve requests sent to peer URLs of the local member, which helps
+// remote peers to connect with local member.
+// User needs to call Start before calling other functions, and call
+// Stop when the Transport is no longer used.
+type Transport struct {
+	RoundTripper http.RoundTripper  // roundTripper to send requests
+	ID           types.ID           // local member ID
+	ClusterID    types.ID           // raft cluster ID for request validation
+	Raft         Raft               // raft state machine, to which the Transport forwards received messages and reports status
+	ServerStats  *stats.ServerStats // used to record general transportation statistics
+	LeaderStats  *stats.LeaderStats // used to record transportation statistics with followers when performing as leader in raft protocol
```

Comment:
The comment is too long. I would suggest to move the comment on top of the field instead of in one line.